rackcity/models/asset.py

```diff
@@ -121,13 +121,34 @@ def validate_owner(value):
         )
 
 
+def validate_location_type(model, rack, rack_position, chassis, chassis_slot):
+    if model.is_rackmount():
+        if not rack or not rack_position:
+            raise ValidationError(
+                "Rackmount assets must have a rack and rack position. "
+            )
+            # WRONG! What if they're offline?! You fool
+        if chassis or chassis_slot:
+            raise ValidationError(
+                "Rackmount assets must not have a chassis or chassis slot. "
+            )
+    else:
+        if not chassis or not chassis_slot:
+            raise ValidationError("Blade assets must have a chassis and chassis slot. ")
+            # WRONG! What if they're offline?! You fool
+        if rack or rack_position:
+            raise ValidationError(
+                "Rackmount assets must not have a rack or rack position. "
+            )
+
+
 class AssetID(models.Model):
     id = models.AutoField(primary_key=True)
 
 
 class AbstractAsset(AssetID):
-    rack_position = models.PositiveIntegerField()
-
+    rack_position = RCPositiveIntegerField(null=True, blank=True,)
```

Comment:
whats the diff between RCPositiveIntegerField and PostiveIntegerField?

Reply:
Same as above - handles empty strings better, which is only necessary now that it's optional